incubator/java-openliberty/image/project/validate.sh

```diff
@@ -54,39 +51,30 @@ the parent version in pom.xml, and test your changes.
   <parent>
     <groupId>${a_groupId}</groupId>
     <artifactId>${a_artifactId}</artifactId>
-    <version>${a_range}</version>
+    <version>${a_version}</version>
     <relativePath/>
   </parent>
   "
   exit 1
 fi
 
-# Skip check below, not sure if we should fix or just remove.   It doesn't account for the fact that
-# the stack now uses pluginManagement, so we could fix to allow this as an acceptable usage too... but what if Jane introduces a profile?
-exit 0
 
-# Check child pom for required liberty version, groupID and artifactId
-l_groupId=$(xmlstarlet sel -T -N x="http://maven.apache.org/POM/4.0.0" -t -v "/x:project/x:build/x:plugins/x:plugin[x:artifactId='liberty-maven-plugin']/x:configuration/x:assemblyArtifact/x:groupId" pom.xml)
-l_artifactId=$(xmlstarlet sel -T -N x="http://maven.apache.org/POM/4.0.0" -t -v "/x:project/x:build/x:plugins/x:plugin[x:artifactId='liberty-maven-plugin']/x:configuration/x:assemblyArtifact/x:artifactId" pom.xml)
-l_version=$(xmlstarlet sel -T -N x="http://maven.apache.org/POM/4.0.0" -t -v "/x:project/x:build/x:plugins/x:plugin[x:artifactId='liberty-maven-plugin']/x:configuration/x:assemblyArtifact/x:version" pom.xml)
+# Check that the child pom has not overridden the Open Liberty version
+child_ol_version_property=$(xmlstarlet sel -T -N x="http://maven.apache.org/POM/4.0.0" -t -v "/x:project/x:properties/x:version.openliberty-runtime" pom.xml)
```

Comment:
SK1:  I like the direction you headed in, and I get it now.

Perhaps all we need to look at for now is whether there is a **runtimeArtifact** at all?   I don't see that allowing  the element **runtimeArtifact** but then requiring it to have empty child elements is OK..Jane has gone astray.    

Note there are several other [methods](https://github.com/OpenLiberty/ci.maven/blob/master/docs/installation-configuration.md) for installing liberty that you should also NOT be doing, that we COULD check for.

I wish we could wait until we've gotten the install down and read  `lib/versions/openliberty.properties` to see what version's installed.  Of course...what if the config later points us somewhere else?   IDK.   

I don't think this is adding that much value over the enforcer OL version property check...but it does seem like it's the most likely thing for Jane to try to do that shouldn't be allowed (by copying/pasting from the parent POM and tweaking the version), so I'm OK with it.

Reply:
Good points.  The intent with the "empty" checks was to make sure those *runtimeArtifact* elements weren't set to begin with. I hadn't thought that Jane could define them but keep them empty which would be strange. 

Yea I think the general idea was to cover the gap of Jane accidentally (or misguidedly) adding her own runtime artifact and not realizing that is not ok to do. This would flag that scenario, although as you said, there are other methods for installing liberty not covered with this. 

Since you are ok with this, lets leave it in and if we want we can revisit how to make this a bit more robust. Or leave it to Champ to do as he chooses.